Simon-Initiative/oli-torus
# Prevent 3rd party cookie problems when inside iframe in delivery

Firefox and Safari do not allow setting cookies on a different host from within an iframe. Courses delivered with Canvas should use the "launch in a new tab" option for viewing course material, but this PR adds a check to see if the course was launched inside of an iframe and replaces the course outline with a link to expand the iframe into the full parent window if so.

I thought about doing this within the "react component" framework we have set up in `components.tsx` and webpack, but it didn't seem worth the trouble and indirection of linking together multiple pieces just to introduce a button. So it's a few lines of vanilla javascript.

To test:
* Delivery a local course through canvas in firefox or safari
* Change the delivery option to launch in an iframe
* Open the course outline within the iframe, make sure the button shows up to launch into the parent window
* Make sure the button works by replacing the canvas parent window with the iframe child
* Make sure activity answer selections work
* Clear cookies
* Re-deploy with the "launch in a new tab" option selected
* Make sure the course outline shows instead of the button and activity answer selection still works
